ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -511,8 +511,13 @@ + (void)updateViewController:(UIViewController *)vc
                                                                              action:nil];
   [backBarButtonItem setMenuHidden:config.disableBackButtonMenu];
 
+  auto isBackButtonCustomized = !isBackTitleBlank || config.disableBackButtonMenu || NO;
+
   if (config.isBackTitleVisible) {
-    if (config.backTitleFontFamily || config.backTitleFontSize) {
+    if ((config.backTitleFontFamily &&
+         ![config.backTitleFontFamily isEqual:@"System"]) ||
```

Comment:
Why do we need to check the font family here?

Reply:
It's because due to the current implementation of `@react-navigation/native-stack`, `backTitleFontFamily` will always be [assigned to the font family of `fonts.regular`](https://github.com/react-navigation/react-navigation/blob/ce6d885/packages/native-stack/src/views/HeaderConfig.tsx#L75-L80), which [will be "System" if the default theme is used](https://github.com/react-navigation/react-navigation/blob/ce6d88559e4a1afeafa84fc839892bb846349d67/packages/native/src/theming/fonts.tsx#L28-L31).

It seems that the "System" font family is already the default on iOS, as setting the font family to "System" does not change the appearance, but in this case, will make `config.backTitleFontFamily` always resolve to `YES`.

Reply:
I see. Do you know, if we could rely on the native component that represents the title of back button, instead of comparing `config.backTitleFontFamily`?

Reply:
Thanks for the PR, this is great!

Can we have this comment in the code please?

Reply:
@zetavg I'm wondering if it would be possible to get title component of back title and check if it has default font set, but I'm afraid this could lead to some flickers (setting the font and other properties -> checking the font of the native component -> setting system back 😄), so I guess we can stay with the current solution.

Reply:
Thanks for explaining, I think I got your point! I do think the current solution of comparing the font against `"System"` isn't perfect, one way is that there may be different ways to use the same system font (maybe `".AppleSystemUIFont"`? I'm not sure), and if `react-navigation` changes to also assign a default value to `config.backTitleFontSize` this will break again.

Your solution inspired me to think of a possible way to get this native behavior to work while still customizing the back button (such as changing the font or customizing the shortened "Back" text) - measure the width of the `headerTitle` and `headerRight` to see if there's enough space on layout, and replace the `backButtonTitle` with the shortened text or hide it. Well, flickering may be a problem.